static/src/javascripts/projects/common/modules/ui/dropdowns.js

```diff
@@ -23,11 +23,63 @@ const updateAria = (container: Element): void => {
 const init = (): void => {
     bean.on(document.body, 'click', `.${buttonCN}`, (e: Event) => {
         const container = (e.currentTarget: any).closest(containerSelector);
+
         if (container) {
-            fastdom.write(() => {
-                container.classList.toggle('dropdown--active');
-                updateAria(container);
-            });
+            const content = container.querySelector(`.${contentCN}`);
+            const isActive: boolean = container.classList.contains(
+                'dropdown--active'
+            );
+            const isAnimated: boolean = container.classList.contains(
+                'dropdown--animated'
+            );
+            const contentEstimatedHeight =
+                content.offsetHeight !== undefined &&
+                content.offsetHeight < window.innerHeight
+                    ? content.offsetHeight
+                    : window.innerHeight;
```

Comment:
Perhaps we can wrap the above in a `fastdom.read` to minimise layout thrashing
```js
fastdom.read(() => {
   const container = ...
   const content = ...

   return {content, isActive, isAnimated ... }
}).then(({content, isActive, isAnimated ... }) => {
  if (isAnimated ...) {
    ...
  }
})
```
Let me know if you'd like some help refactoring

Reply:
thanks! i think you pretty much nailed it with the example although i don't think our version of fastdom has promises in it. (does it?) i'm about to commit it wrapped in .read() though

Reply:
Oh good shout, you'll need to import `lib/fastdom-promise`, which is our promise wrapper, rather than just vanilla `fastdom`. Currently, there's a `fastdom.write()` inside the `fastdom.read()` which confuses `fastdom` a bit..

Reply:
haha oops, alright! let me get back to you on this then